controller/src/main/java/io/pravega/controller/server/ControllerServiceStarter.java

```diff
@@ -184,6 +177,23 @@ protected void startUp() {
                         .thenAcceptAsync(x -> controllerEventProcessors.startAsync(), controllerExecutor);
             }
 
+            // Setup and start controller cluster listener after all sweepers have been initialized.
+            if (serviceConfig.isControllerClusterListenerEnabled()) {
+                cluster = new ClusterZKImpl((CuratorFramework) storeClient.getClient(), ClusterType.CONTROLLER);
+                List<FailoverSweeper> failoverSweepers = new ArrayList<>();
+                failoverSweepers.add(taskSweeper);
+                failoverSweepers.add(txnSweeper);
+                if (serviceConfig.getEventProcessorConfig().isPresent()) {
+                    assert controllerEventProcessors != null;
```

Comment:
If `controllerEventProcessors` is null at this point, then the expected behavior is that the controller exits? As I understand it, the only possibility for it to be null is if `serviceConfig.getEventProcessorConfig().isPresent()` is false, is it right?